File: programmers/java/src/level1/SumOfDivisor.java
package level1;

import java.util.stream.IntStream;

public class SumOfDivisor {
    public static void main(String[] args) {
        System.out.println(solution(12));
        System.out.println(solution(5));
        System.out.println(solution(4));
    }

    public static int solution(int n) {
        int answer = IntStream.range(1, (int) Math.sqrt(n) + 1)
                .filter(x -> n % x == 0)
                .reduce(0, (sum, x) -> sum + (x + n / x));

        if ((int) Math.sqrt(n) == Math.sqrt(n)) {
            answer -= Math.sqrt(n);
        }
        return answer;
    }
}

File: bongf/python/level1/sum_of_divisor.py
def solution(n):
    answer = 0
    for i in range (1, (int)(n** 0.5)+1) :
        if n % i == 0 :
            answer += i
            if ( n//i != i ):
                answer += (n // i) 
    return answer

# print(solution(12))
print(solution(16))

